index.js

```diff
@@ -1790,6 +1790,30 @@
   S.unlines =
   def('unlines', [List], R.compose(R.join(''), R.map(R.concat(_, '\n'))));
 
+
+  //# unfoldr :: (b -> Maybe (a, b)) -> b -> [a]
+  //.
+  //. The unfoldr function is a `dual' to foldr: while foldr reduces a list to
+  //. a summary value, unfoldr builds a list from a seed value. The function
+  //. takes the element and returns Nothing if it is done producing the list or
+  //. returns Just (a,b), in which case, a is a prepended to the list and b is
+  //. used as the next element in a recursive call.
+  //.
+  //. ```Javascript
+  //. > S.unfoldr(function(n){return n >= 5 ? S.Nothing() : S.Just([n, n + 1]);}, 1)
+  //. [1, 2, 3, 4]
+  //. ```
+  S.unfoldr = def('unfoldr', [Function, a], function(f, init) {
+    var resultList = [];
+    var maybeTuple = f(init);
+    while (is(Just, maybeTuple)) {
+      var tuple = fromMaybe(null, maybeTuple);
+      resultList.push(tuple[0]);
+      maybeTuple = f(tuple[1]);
```

Comment:
Let's use `maybeTuple.value` rather than `fromMaybe(null, maybeTuple)`. While others shouldn't assume the existence of a `value` property on a Just, it's acceptable for Sanctuary itself to do so (and it *must*, in places).

```javascript
resultList.push(maybeTuple.value[0]);
maybeTuple = f(maybeTuple.value[1]);
```

It'd be nice to rename `maybeTuple` something short such as `m`.